src/libraries/System.Net.Http/src/System/Net/Http/HttpTelemetry.cs

```diff
@@ -36,10 +37,25 @@ internal sealed class HttpTelemetry : EventSource
         // - A stop event's event id must be next one after its start event.
 
         [Event(1, Level = EventLevel.Informational)]
-        public void RequestStart(string scheme, string host, int port, string pathAndQuery, int versionMajor, int versionMinor)
+        private void RequestStart(string scheme, string host, int port, string pathAndQuery, byte versionMajor, byte versionMinor, HttpVersionPolicy versionPolicy)
         {
             Interlocked.Increment(ref _startedRequests);
-            WriteEvent(eventId: 1, scheme, host, port, pathAndQuery, versionMajor, versionMinor);
+            WriteEvent(eventId: 1, scheme, host, port, pathAndQuery, versionMajor, versionMinor, versionPolicy);
+        }
+
+        [NonEvent]
+        public void RequestStart(HttpRequestMessage request)
+        {
+            Debug.Assert(request.RequestUri != null);
+
+            RequestStart(
+                request.RequestUri.Scheme,
+                request.RequestUri.IdnHost,
+                request.RequestUri.Port,
+                request.RequestUri.PathAndQuery,
+                (byte)request.Version.Major,
+                (byte)request.Version.Minor,
```

Comment:
Are we okay truncating these if user sets them to something crazy? Seems like something they'd want to see in diagnostics.

Reply:
I assumed we throw for anything unknown, do we allow it?

Reply:
Not really. We allow it and we just connect over what we have (depending on `VersionPolicy`).
Minor is not checked almost at all (except for 1.0 and chunked content ASAIK).

Reply:
What was the motivation to change to byte here?